ngi_pipeline/database/filesystem.py

```diff
@@ -160,9 +166,18 @@ def create_charon_entries_from_project(project, best_practice_analysis="whole_ge
                                      'seqrun "{}" already exists; next...'.format(project, sample,
                                                                                   libprep, seqrun))
                     else:
+                        update_failed=True
                         LOG.error(e)
                         continue
 
+    if update_failed :
+        if retry_on_fail:
+        create_charon_entries_from_project(project, best_practice_analysis=best_practice_analysis,
+                                       sequencing_facility=sequencing_facility
+                                       force_overwrite=force_overwrite, delete_existing=delete_existing,
+                                       retry_on_fail=False):
```

Comment:
Agree, that was the first google hit I found yesterday while travelling back from uppsala... Maybe there is something more standard. I tried to think about possible race conditions when multiple instances are called and I am not able to find any... but still I am not 100% sure this recursive call is always doing what is expected